t/shared.t

```diff
@@ -33,5 +32,23 @@ is( $hash{multi}->{key1},      "foo", "multidimension, key1" );
 is( $hash{multi}->{arr1}->[0], "bar", "multidimension, arr1 - key0" );
 is( $hash{multi}->{arr1}->[1], "baz", "multidimension, arr1 - key1" );
 
-unlink("vars.db");
-unlink("vars.db.lock");
+{
+    @array = (0);
+    my @pids;
+    for my $i (0..5) {
+        my $pid = fork();
+        if ($pid == 0) {
+            # child
+            sleep 1;
```

Comment:
Ok, I tried polling the children with sigzero (another signal I've never used).  
```
{
    @array = (0);
    my $parent = $$;
    print "parent: $$\n";
    my @pids;
    my $ready_pids;
    $SIG{CONT} = sub { print "child $$\n"; $ready_pids++ };

    for my $i (0..5) {
        my $pid = fork();
        if ($pid == 0) {
            # child
            POSIX::pause();
            print "after pause\n";
            push @array, 1;
            exit 0;
        }
        push @pids, $pid;
    }

    while (1) {
        my $success = kill ZERO => @pids;
        last if $success == 6; # the 6 children are ready to go
    }

    my $foo = kill 'CONT' => @pids;
    print "CONT: $foo\n";

    waitpid $_, 0 for @pids;

    cmp_deeply \@array, [qw/0 1 1 1 1 1 1/], 'race condition avoided';
}
```
It almost works, but it still hangs 30% of the time.  If I `sleep1; kill CONT => @pids` a second time, it doesn't hang.  So somehow sigzero is still too early I guess?